app/services/users/submit-profile-details.service.js

```diff
@@ -0,0 +1,88 @@
+'use strict'
+
+/**
+ * Orchestrates validating the data for `/users/me/profile-details` page
+ * @module SubmitProfileDetailsService
+ */
+
+const UserModel = require('../../models/user.model.js')
+const ProfileDetailsValidator = require('../../validators/users/profile-details.validator.js')
+
+const navigationLinks = [
+  { active: true, href: '/system/users/me/profile-details', text: 'Profile details' },
+  { href: '/account/update-password', text: 'Change password' },
+  { href: '/signout', text: 'Sign out' }
+]
+
+/**
+ * Orchestrates validating the data for `/users/me/profile-details` page
+ *
+ * It first retrieves the session instance for the returns requirements journey in progress.
+ *
+ * The user input is then validated and the site description in the payload is saved in the session. If there is a
+ * validation error the controller will re-render the page. If all is well the controller will redirect to the next page
+ * in the journey.
+ *
+ * @param {number} userId - The ID of the user
+ * @param {object} payload - The submitted form data
+ * @param {object} yar - The Hapi `request.yar` session manager passed on by the controller
+ *
+ * @returns {Promise<object>} The page data for the site description page including any validation error details
+ */
+async function go(userId, payload, yar) {
+  const validationResult = _validate(payload)
+
+  if (!validationResult) {
+    await _save(userId, payload)
+
+    yar.flash('notification', {
+      title: 'Updated',
+      text: 'Profile details saved'
+    })
```

Comment:
In the existing system, when the profile details are successfully updated, the user is just returned to the home screen with no notification of whether the details have been updated.

I've added this notification and currently the controller keeps the user on this same page so they see the notification.

Not sure if that is a valid change to the journey.

Reply:
As noted above, this is ideal for now (and better than the legacy solution).